src/core/Akka/Util/Result.cs

```diff
@@ -137,7 +137,30 @@ public static Result<T> Failure<T>(Exception exception)
         /// <returns>TBD</returns>
         public static Result<T> FromTask<T>(Task<T> task)
         {
-            return task.IsCanceled || task.IsFaulted ? new Result<T>(task.Exception) : new Result<T>(task.Result);
+            if(!task.IsCompleted)
+                throw new ArgumentException("Task is not completed. Result.FromTask only accepts completed tasks.", nameof(task));
```

Comment:
New behavior, `.FromTask()` did not check if the `Task` argument is completed or not, it should only accept completed tasks